src/dispatch/plugins/dispatch_slack/middleware.py

```diff
@@ -27,33 +27,33 @@
 
 
 @timer
-def resolve_context_from_conversation(
-    channel_id: str, thread_id: Optional[str] = None
-) -> Optional[Subject]:
-    """Attempts to resolve a conversation based on the channel id or message_ts."""
+def resolve_context_from_conversation(channel_id: str) -> Optional[Subject]:
+    """Attempts to resolve a conversation based on the channel id."""
     db_session = SessionLocal()
     organization_slugs = [o.slug for o in organization_service.get_all(db_session=db_session)]
     db_session.close()
+
     for slug in organization_slugs:
         scoped_db_session = refetch_db_session(slug)
 
         conversation = conversation_service.get_by_channel_id_ignoring_channel_type(
-            db_session=scoped_db_session, channel_id=channel_id, thread_id=thread_id
+            db_session=scoped_db_session, channel_id=channel_id
         )
+
         if conversation:
-            if thread_id:
+            if conversation.incident:
                 subject = SubjectMetadata(
-                    type="case",
-                    id=conversation.case_id,
+                    type="incident",
+                    id=conversation.incident_id,
                     organization_slug=slug,
-                    project_id=conversation.case.project_id,
+                    project_id=conversation.incident.project_id,
                 )
             else:
                 subject = SubjectMetadata(
-                    type="incident",
-                    id=conversation.incident_id,
+                    type="case",
+                    id=conversation.case_id,
```

Comment:
I added some Subject type enums we could leverage here:

https://github.com/Netflix/dispatch/blob/master/src/dispatch/plugins/dispatch_slack/models.py#L52-L58

Although, these should probably be moved to the top level `enum.py` file as they're not specific to Slack.

Reply:
Ok, I'll address it in a separate PR.